validation/coupled_models/coupled_atmosphere_ocean_model.jl

```diff
@@ -0,0 +1,75 @@
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Grids: topology
+
+#####
+##### Utilities
+#####
+
+using Oceananigans.Models: AbstractModel
+
+import Oceananigans.TimeSteppers: time_step!, update_state!
+import Oceananigans: fields
+
+struct CoupledAtmosphereOceanModel{O, A, C, P} <: AbstractModel{Nothing}
+    atmos :: A
+    ocean :: O
+    clock :: C
+    air_sea_flux_parameters :: P
+end
+
+function CoupledAtmosphereOceanModel(atmos, ocean; air_sea_flux_parameters=(; cᴰ=2e-3, ρ_atmos=1, ρ_ocean=1024))
+    clock = atmos.clock
+    return CoupledAtmosphereOceanModel(atmos, ocean, clock, air_sea_flux_parameters)
+end
+
+fields(model::CoupledAtmosphereOceanModel) = fields(model.atmos) # convenience hack for now
+
+function update_state!(coupled_model::CoupledAtmosphereOceanModel, update_atmos_ocean_state=true)
+    atmos_model = coupled_model.atmos
+    ocean_model = coupled_model.ocean
+
+    if update_atmos_ocean_state
+        update_state!(ocean_model)
```

Comment:
If the models are coupled, which I presume they should be, then this means we treat the terms from the other model explicitly.  This looks good as I don't suppose we would ever want to treat these terms implicitly, but saying something in case someone has any ideas if this might be a concern.

Reply:
Can you clarify what you mean? Do you mean we could impose fluxes into the ocean when stepping between n and n+1 using the atmospheric state at n+1?